presto-main/src/main/java/com/facebook/presto/sql/planner/LogicalPlanner.java

```diff
@@ -128,11 +131,42 @@ private RelationPlan createInsertPlan(Analysis analysis)
 
         TableMetadata tableMetadata = metadata.getTableMetadata(session, target);
 
+        List<String> columnNames = getVisibleColumnNames(tableMetadata);
+        List<ColumnMetadata> columns = getVisibleColumns(tableMetadata);
+
+        RelationPlan plan = createRelationPlan(analysis);
+
+        Optional<List<String>> insertColumns = analysis.getInsertColumns();
+        if (insertColumns.isPresent() && !insertColumns.get().equals(columnNames)) {
+            PlanNode source = plan.getRoot();
+
+            Map<Symbol, Expression> assignments = columns.stream()
+                    .filter(column -> !column.isHidden())
```

Comment:
this .filter is not needed (you have only visible columns on `columns` variable)